lms/djangoapps/course_api/blocks/tests/test_views.py

```diff
@@ -414,3 +414,8 @@ def test_invalid_course_id(self):
 
     def test_non_existent_course(self):
         self.verify_response(403, params={'course_id': str(CourseLocator('non', 'existent', 'course'))})
+
+    def test_non_existent_course_anonymous(self):
+        self.client.logout()
+        self.query_params['username'] = ''
+        self.verify_response(403, params={'course_id': str(CourseLocator('non', 'existent', 'course'))})
```

Comment:
Confirmed that this test reproduced the ValueError we were seeing in production.